sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilters.scala

```diff
@@ -778,6 +778,38 @@ class ParquetFilters(
           )
         }
 
+      case sources.StringEndsWith(name, prefix)
+        if pushDownStringPredicate && canMakeFilterOn(name, prefix) =>
+        Option(prefix).map { v =>
+          FilterApi.userDefined(binaryColumn(nameToParquetField(name).fieldNames),
+            new UserDefinedPredicate[Binary] with Serializable {
+              private val strToBinary = Binary.fromReusedByteArray(v.getBytes)
```

Comment:
You can just keep `UTF8String.fromBytes(strToBinary.getBytes)` instead doing it every time in `keep`.

Reply:
good catch. updated